src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9268,7 +9268,11 @@ void Unit::UpdateSpeed(UnitMoveType mtype)
 
     if (float minSpeedMod = (float)GetMaxPositiveAuraModifier(SPELL_AURA_MOD_MINIMUM_SPEED))
     {
-        float min_speed = minSpeedMod / 100.0f;
+        float baseMinSpeed = 1.0f;
+        if (!(GetOwnerGUID().IsPlayer() || IsHunterPet()) && GetTypeId() == TYPEID_UNIT)
```

Comment:
please, dont try putting the negation outside, it makes the code hard to decipher